### PR TITLE
Priy/update show coordinates sample

### PR DESCRIPTION
**Pull request by prupani-7**
## Description

PR to update the sample to use composable mapview from the toolkit
[Versions.gradle](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/148/files#diff-d737df1dc5c860aab6f96d16a591f7d245e544d0537751fea8f139fa7abcf52c) file from [PR ](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/148) uses "arcgisToolkitVersion = '200.3.0-1'"

So, [update new module script PR](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/148) should be merged before this PR

## Links and Data

Sample Epic: `runtime/kotlin/issues/3368`

## What To Review

-  Review updated files

## How to Test

Run the sample on the sample viewer or the repo.

### CODE REVIEW

**Review by hud10837 — COMMENTED** [review]
@prupani-7 Looks good and works as expected, a couple small comments

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt [hunk 1]
```diff
@@ -16,40 +16,93 @@
...
+    )
+    graphicsOverlay.graphics.add(coordinateLocation)
+    // update the coordinate notations using the initial point
+    coordinateLocation.geometry = initialPoint
+    mapViewModel.toCoordinateNotationFromPoint(initialPoint)
 
     Scaffold(
         topBar = { SampleTopAppBar(title = sampleName) },
-        content = {
+        content = { it ->
```
[the remark this answers is not in the document]
Comment (prupani-7) [reply show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt]: removed it, thanks!

**Comment by prupani-7** [comment]
@hud10837 thanks for the feedback! addressed them, re-review?

**Review by hud10837 — APPROVED** [review]
LGTM! 👍

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt
Comment (shubham7109) [inline show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt:1]: The graphic point does not update when I try the reverse workflow. 
- Tapping a point elsewhere
- Pasting in a predeterminate coordinate 
- All text fields update to the new coordinate as expected
- The graphic point on map does not

Could you debug to see if this issue is due to the toolkit MapView or the sample workflow?

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt [hunk 2]
```diff
@@ -35,21 +49,60 @@ import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.Ma
...
-                ComposeMapView(
+                MapView(
                     modifier = Modifier.fillMaxSize(),
-                    mapViewModel = mapViewModel,
-                    onSingleTap = { singleTapConfirmedEvent ->
-                        mapViewModel.onMapTapped(singleTapConfirmedEvent.mapPoint)
+                    arcGISMap = arcGISMap,
+                    graphicsOverlays = graphicsOverlayCollection.apply {
+                        this.add(graphicsOverlay)
+                    },
```
Comment (shubham7109) [inline show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt:90]: I think simple assignment properties here more clear, and we can move the `apply` block away from the Compose call and add it to the line the val is declared. 

```kotlin
    // graphics overlay for the MapView to draw the graphics
    val graphicsOverlay = remember { GraphicsOverlay() }
    // the collection of graphics overlays used by the MapView
    val graphicsOverlayCollection = rememberGraphicsOverlayCollection().apply {
        add(graphicsOverlay)
    }
```

```suggestion
                    graphicsOverlays = graphicsOverlayCollection,
```

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt [hunk 3]
```diff
@@ -35,21 +49,60 @@ import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.Ma
...
+                    onSingleTapConfirmed = { singleTapConfirmedEvent ->
+                        /**
+                         * Updates the tapped graphic and coordinate notations using the [tappedPoint]
+                         */
```
Comment (shubham7109) [inline show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt:94]:
```suggestion
                        // retrieve the map point on MapView tapped
```

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt [hunk 4]
```diff
@@ -35,21 +49,60 @@ import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.Ma
...
+    val basemapLayer = ArcGISTiledLayer(LocalContext.current.applicationContext.getString(R.string.basemap_url))
+    val arcGISMap = ArcGISMap(Basemap(basemapLayer))
+    // the collection of graphics overlays used by the MapView
+    val graphicsOverlayCollection = rememberGraphicsOverlayCollection()
+    // graphics overlay for the MapView to draw the graphics
+    val graphicsOverlay = remember { GraphicsOverlay() }
+    // set up a graphic to indicate where the coordinates relate to, with an initial location
+    val initialPoint = Point(0.0, 0.0, SpatialReference.wgs84())
+
+    val coordinateLocation = Graphic(
```
Comment (shubham7109) [inline show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt:62]: `coordinateLocation` might be confused with an actual location point. 

Could rename it to `coordinateLocationGraphic`
```suggestion
    val coordinateLocationGraphic = Graphic(
```

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt
Comment (shubham7109) [inline show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt:1]: The app crashes when I type in an incorrect coordinate point. Ideally, it should display the `messageDialogVM.showMessageDialog`

**Review by shubham7109 — COMMENTED** [review]
Nice changes @prupani-7. Few comments to address..

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt
Comment (prupani-7) [reply show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt:1]: good catch, fixed...
the geometry of the coordinateLocationGraphic was not getting updated to the new Point.

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt (hunk 2 above)
Comment (prupani-7) [reply show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/MainScreen.kt]: This fixed the issue with showMessageDialog not showing and app crashing with error (GO object already owned)
I think everytime MapView composable was called, we were adding the same GO object to the GO collection, without clearing it. 
but moving the apply block in declaration fixes this. Thanks!

File: show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt
Comment (prupani-7) [reply show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/components/MapViewModel.kt:1]: see above comment..

**Comment by prupani-7** [comment]
@shubham7109 thanks for the feedback, applied your suggestions, please re-review?

**Review by shubham7109 — APPROVED** [review]
Looks good, thanks for applying the changes 👍

**Pull request merged**